src/drivers/net/bnxt/bnxt.c

```diff
@@ -2226,27 +2655,44 @@ static void bnxt_service_nq ( struct net_device *dev )
 		bnxt_db_nq ( bp );
 }
 
+static void bnxt_er_timer ( struct retry_timer *timer, int over __unused )
+{
+	struct bnxt *bp = container_of (timer, struct bnxt, timer );
+
+	/* Restart timer */
+	start_timer_fixed ( timer, BNXT_ER_TIMER_INTERVAL ( bp ) );
+	if ( bp->er.driver_initiated_recovery ) {
+		bnxt_er_task ( bp, 1 );
+	}
+}
+
 static void bnxt_poll ( struct net_device *dev )
 {
+	struct bnxt *bp = dev->priv;
+
 	mb (  );
 	bnxt_service_nq ( dev );
 	bnxt_service_cq ( dev );
+
+	if ( bp->er.er_initiate ) {
+		bnxt_er_task ( bp, 0 );
+		bp->er.er_initiate = 0;
+	}
+
 }
 
 static void bnxt_close ( struct net_device *dev )
 {
 	struct bnxt *bp = dev->priv;
 
 	DBGP ( "%s\n", __func__ );
-	bnxt_down_nic (bp);
+	if ( bp->er.driver_initiated_recovery ) {
+		stop_timer ( &bp->timer );
+	}
```

Comment:
Minor point, but this can (and probably should) be an unconditional `stop_timer()`, just so that nobody has to worry about whether or not it's possible for the timer to be left running inadvertently.  (It is safe to call `stop_timer()` on an already-stopped timer.)